ssh/ssh.go

```diff
@@ -0,0 +1,61 @@
+package ssh
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"golang.org/x/crypto/ssh"
+	"log"
+	"strings"
+
+	"github.com/SUSE/scf-secret-generator/model"
+)
+
+type SSHKey struct {
+	PrivateKey  string // Name to associate with private key
+	Fingerprint string // Name to associate with fingerprint
+}
+
+func GenerateSSHKey(secretData map[string][]byte, key SSHKey) {
+	// generate private key
+	private, err := rsa.GenerateKey(rand.Reader, 4096)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	privateBlock := pem.Block{
+		Type:    "RSA PRIVATE KEY",
+		Headers: nil,
+		Bytes:   x509.MarshalPKCS1PrivateKey(private),
+	}
+
+	// PEM encode private key
+	secretKey := strings.Replace(strings.ToLower(key.PrivateKey), "_", "-", -1)
+	secretData[secretKey] = pem.EncodeToMemory(&privateBlock)
+
+	// generate MD5 fingerprint
+	public, err := ssh.NewPublicKey(&private.PublicKey)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fingerprintKey := strings.Replace(strings.ToLower(key.Fingerprint), "_", "-", -1)
+	secretData[fingerprintKey] = []byte(ssh.FingerprintLegacyMD5(public))
+}
+
+func ParseSSHKey(keys map[string]SSHKey, configVar *model.ConfigurationVariable) {
+	var key SSHKey
+
+	if _, ok := keys[configVar.Generator.ID]; ok {
```

Comment:
Why is this condition necessary?

> An attempt to fetch a map value with a key that is not present in the map will return the zero value for the type of the entries in the map.

So the `var` declaration could be completely eliminated with:

```
key := keys[configVar.Generator.ID]
```

And maybe add a comment that the key might not exist yet at that point.

Reply:
Fixed in e61d632